java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -257,6 +160,8 @@ public <RequestT, ResponseT, ErrorT> ResponseT performRequest(
             } catch (InterruptedException e) {
                 throw new IOException("HttpRequest was interrupted", e);
             }
+        } else {
```

Comment:
We probably should do this check in constructor to refuse the creation of the transport for non-supported clients. May be we should have 2 constructors:

```
 public AwsSdk2Transport(
            @CheckForNull SdkAsyncHttpClient asyncHttpClient,
            @Nonnull String host,
            @Nonnull String signingServiceName,
            @Nonnull Region signingRegion,
            @CheckForNull AwsSdk2TransportOptions options) {
```

```
 public AwsSdk2Transport(
            @CheckForNull SdkHttpClient syncHttpClient,
            @Nonnull String host,
            @Nonnull String signingServiceName,
            @Nonnull Region signingRegion,
            @CheckForNull AwsSdk2TransportOptions options) {
```
To prevent the possibility to use non compatible clients?

Reply:
Good idea, updated.